.buildkite/collect-bonds.yml

```diff
@@ -0,0 +1,50 @@
+agents:
+  queue: "snapshots"
+
+steps:
+  - label: ":hammer_and_wrench: :rust: Build"
+    commands:
+    - '. "$HOME/.cargo/env"'
+    - 'cargo build --release --bin collect --bin store'
+    artifact_paths:
+      - target/release/collect
+      - target/release/store
+
+  - wait: ~
+
+  - label: ":gear: Setup"
+    commands:
+    - '. "$HOME/.cargo/env"'
+
+  - wait: ~
+
+  - command: echo "--> Start of concurrency gate"
+    concurrency_group: 'validator-bonds/collect-bonds'
+    concurrency: 1
+
+  - wait: ~
+
+  - label: ":microscope: Collect Bonds"
+    commands:
+    - 'buildkite-agent artifact download --include-retried-jobs target/release/collect .'
+    - 'chmod +x target/release/collect'
+    - |
+      ./target/release/collect \
```

Comment:
this binary name is a bit too generic - especially as it there is other collect jobs, the same goes for  `store`

I suggest to convert collect to validator-bonds-cli or similar (this could be used to get other data from chain or do whatever on-chain).

I also suggest to make store a new binary inside the api crate.

Then if you change api's data::data to repositories::bond, all functions related to loading/storing DB data of the bonds would go there; the new binary in the api crate would call the code inside the crate::repositories::bond to store the data - making for a bit more concise code